### ISSUE TITLE
Cannot restore backup.ab - md5 does not match 

### ISSUE THREAD

**Issue by atpanos** [TeamWin/Team-Win-Recovery-Project#1126]
Hello, i have a problem with twrp 3.1.1-0 on moto g5 plus (potter)

after i put in cmd 'adb restore backup.ab i get following output:

```
command is: 'adbrestore' and there is no value
Wiping System
Formatting System using make_ext4fs...
Restoring System...
[System done (181 seconds)]
E:md5 doesn't match!
E: file md5: <some md5 hash>
E: check md5: <other md5 hash>
Restore complete.
```

Tried different other TWRP versions but none of them can decrypt the partition.

**Comment by stephendt** [comment]
I am experiencing the same issue on my Xiaomi Redmi Note 4 (Mido) on TWRP 3.2.1.0.

**Comment by ghost** [comment]
Same issue, Xiaomi Mi 5 over TWRP 3.2.1-1

Command executed in PC:
`adb restore <twrp_backup_name>`

Log over TWRP:
```
Restoring Boot...
I:Reading '/tmp/twadbrestore', writing '/dev/block/sde36'
opening TW_ADB_BU_CONTROL
Sending TWEOF to adb backup
[Boot done (5 seconds)]
I:Restoring MD5TRAILER
I:Restoring TWMD5
E:md5 doesn't match!
E:file md5: ff6bf986c6b99c160d2a7eabe0fdae6d
E:check md5: 2c33ce6b9b3c5f8565d3b0aa7287d24c
Error during restore process.
```

The task at hand was to perform a recovery of LineageOS 14.1 over the installed LineageOS 15.1. I had a backup of LineageOS 15.1 which worked without issues, but the 14.1 ones didn't work (MD5 error)

**Comment by elsieholmes** [comment]
Any chance of this crap getting in the near future?

**Comment by burinov** [comment]
Have the same problem here...

**Comment by bigbiff** [comment]
Adb backup issues have been fixed in 3.2.3. Unfortunately older backups are not restorable.
